lib/std/Thread.zig

```diff
@@ -76,6 +76,188 @@ pub fn spinLoopHint() void {
+
+            const err = c.pthread_getname_np(self.data.handle, buffer.ptr, maxNameLen + 1);
+            return switch (err) {
+                0 => std.mem.spanZ(buffer),
+                os.ERANGE => unreachable,
+                else => return os.unexpectedErrno(err),
+            };
+        } else if (self.data.handle == getCurrentId()) {
+            var buffer = try allocator.allocSentinel(u8, maxNameLen, 0);
+
+            const err = try os.prctl(.GET_NAME, .{@ptrToInt(buffer.ptr)});
+            return switch (err) {
+                0 => std.mem.spanZ(buffer),
+                else => return os.unexpectedErrno(err),
+            };
+        } else if (!use_pthreads) {
+            var buf: [32]u8 = undefined;
+            const path = try std.fmt.bufPrint(&buf, "/proc/self/task/{d}/comm", .{self.data.handle});
+
+            const file = try std.fs.cwd().openFile(path, .{});
+            defer file.close();
+
+            const data = try file.reader().readAllAlloc(allocator, maxNameLen + 1);
+            return if (data.len >= 1) data[0 .. data.len - 1] else null;
+        } else {
+            // musl doesn't provide pthread_getname_np and there's no way to retrieve the thread id of an arbitrary thread.
+            return error.Unsupported;
+        },
+        .windows => if (std.Target.current.os.isAtLeast(.windows, .win10_rs1)) |res| {
+            // GetThreadDescription is only available since version 1607, which is 10.0.14393.795
+            // See https://en.wikipedia.org/wiki/Microsoft_Windows_SDK
+            if (!res) {
+                return error.Unsupported;
+            }
+
+            var name_w: os.windows.LPWSTR = undefined;
+            try windows.GetThreadDescription(self.data.handle, &name_w);
+            defer windows.LocalFree(name_w);
+
+            return try std.unicode.utf16leToUtf8Alloc(
+                allocator,
+                std.mem.spanZ(name_w),
+            );
+        } else {
+            return error.Unsupported;
+        },
+        .macos, .ios, .watchos, .tvos => if (use_pthreads) {
+            var buffer = try allocator.allocSentinel(u8, maxNameLen, 0);
+
+            const err = c.pthread_getname_np(self.data.handle, buffer.ptr, maxNameLen + 1);
+            return switch (err) {
+                0 => std.mem.spanZ(buffer),
+                os.ESRCH => unreachable,
+                else => return os.unexpectedErrno(err),
+            };
+        },
+        .netbsd => if (use_pthreads) {
+            var buffer = try allocator.allocSentinel(u8, maxNameLen, 0);
+
+            const err = c.pthread_getname_np(self.data.handle, buffer.ptr, maxNameLen + 1);
+            return switch (err) {
+                0 => std.mem.spanZ(buffer),
+                os.EINVAL => unreachable,
+                os.ESRCH => unreachable,
+                else => return os.unexpectedErrno(err),
+            };
+        },
+        .freebsd => if (use_pthreads) {
```

Comment:
You can merge this with the `openbsd` case below.

Reply:
it was splitted because freebsd function is returning `int` and openbsd function is returning `void` (previously the return value was checked).

but as it, freebsd case doesn't check for `pthread_get_name_np` return value.

Reply:
I opted to keep them split right now mainly to add the FreeBSD comment, but I can also add the comment in the `freebsd, openbsd` case.